### DISCUSSION TITLE
Extreme 3D Pro Joystick

### DISCUSSION THREAD

**Discussion by waterwood45** [winder/Universal-G-Code-Sender#2276] [Help] [answered]
Hi everyone, 

Currently trying to connect the following joystick to the UGS:
-Logitech Extreme 3D Pro
-M/N: J-UK17
-P/N: 863225-1000

In UGS, I activated the 'Activate joystick' option. 

The machine is connected via GRBL, Port: Com3, Bud:115200 and 'Home Machine' has been done. 

Are there any suggestions, what I could try further to make it work?

2 things I have in mind that could be the problem:
1) The joystick is too old.
2) This type of joystick is not supported by UGS.

Thank you very much for your help!

**Comment by breiler** [comment]
When you go to the settings for the gamepad/joystick it should register buttons and movements and highlight them: https://github.com/winder/Universal-G-Code-Sender/wiki/Usage#gamepad-and-joystick

If nothing is highlighted then the controller has probably not been found in the game controller database. Normally, in this case the older is better because the chance of someone registering the mappings for it is higher.

You could try generate a mapping string described here:
https://github.com/gabomdq/SDL_GameControllerDB/#create-new-mappings

Post it here and I'll add it to the database.

**Reply by waterwood45** [reply]
Dear Joacim,

Thank you very much for your help! This is very kind of you!

Below, the mapping string for the joystick: 

030000006d04000015c2000000000000,Logitech Extreme 3D Pro M/N: J-UK17 P/N: 863225-1000,a:b11,b:b10,x:b8,y:b9,back:b6,guide:b7,start:b5,leftstick:b2,rightstick:-a1,leftshoulder:b4,rightshoulder:b0,dpup:h0.1,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,leftx:a0,lefty:a2,platform:Windows, 

All the Best,

**Reply by waterwood45** [reply]
Dear Joacim,

Sorry for this basic question: 
Do I have to download a nightly builds to use the updated database or should it refresh by itself?
Currently, I have win64-ugs-platform-app-2.0.14 and win64-ugs-platform-app-2.0.18 downloaded.

All the Best,

**Reply by breiler** [reply, marked as answer]
Sorry for the delay, it should be available in the latest nightly build here:
https://github.com/winder/Universal-G-Code-Sender#downloads

**Reply by waterwood45** [reply]
Dear Joacim,

Thank you! Works perfectly fine ;-)!
